src/bluecore_api/app/main.py

```diff
@@ -71,21 +55,31 @@ async def scope_mapper(claim_auth: list) -> list:
         authorization_claim="realm_access",
     )
 
-    # Add Keycloak middleware
-    setup_keycloak_middleware(
-        app,
+    keycloak_middleware = KeycloakMiddleware(
+        app=base_app,
         keycloak_configuration=keycloak_config,
-        exclude_patterns=["/docs", "/openapi.json", "/api/docs", "/api/openapi.json"],
         scope_mapper=scope_mapper,
+        exclude_patterns=[],
+    )
+
+    middleware_wrapped_app = BypassKeycloakForGet(
+        app=base_app, keycloak_middleware=keycloak_middleware
     )
+    application = CompatibleFastAPI(app=middleware_wrapped_app)
 
 
-@app.get("/")
+#########################-------------------------------------------------------
```

Comment:
Minor non-block suggestion but these comments that start with `#` could be moved to the function as a __docstring__ following more idiomatic python conventions i.e.

```python
@base_app.get("/")
async def index():
    """Public GET Routes"""
    return {"message": "Blue Core API"}
 ```